src/utils/with-react.js

```diff
@@ -73,4 +79,61 @@ const applyDefaults = ceInst => {
 
 export { applyDefaults };
 
-export default createElement => val(createElement);
+const distributeProperties = ({ children, ...properties }, componentClass) => {
+  // initialize
+  const attrs = {};
+  const props = {};
+  let map;
+  // iterate over all properties
+  Object.keys(properties).forEach(name => {
+    let value = properties[name];
+    // classify property by type to select correct map object
+    // (note that unregistered properties are classified as attr(ibute)s via their undefined .type)
+    const event = name.indexOf('on') === 0 && Function; // Function if onXXX event name, false otherwise
```

Comment:
Clever approach! Although I wonder if it could not be a trap if something else starts with 'on'. I could be wrong though.

Reply:
Tightened it via checking it continues with uppercase letter, making the 'trap' more unlikely as per React onXXX conventions.